src/jvm/main/org/jetbrains/kotlinx/lincheck/util/AnalysisSections.kt

```diff
@@ -93,4 +164,26 @@ internal inline fun <R> runOutsideIgnoredSection(block: () -> R): R {
     } finally {
         descriptor.restoreIgnoredSectionDepth(depth)
     }
-}
\ No newline at end of file
+}
+
+@Suppress("UNUSED_PARAMETER")
+internal fun getDefaultSilentSectionType(className: String, methodName: String): AnalysisSectionType? {
```

Comment:
How do we decide when to enter and leave a silent section? 

Besides, I would place atomic and silent section declarations somewhere in a single place in a separate file, so one can easily observe them.

Reply:
> Besides, I would place atomic and silent section declarations somewhere in a single place in a separate file, so one can easily observe them.

Ok, I agree we should collect this logic in a single place. 
I created a separate issue for this refactoring: https://github.com/JetBrains/lincheck/issues/616

We can do it separately so not to postpone this PR. 
Also, we already briefly discussed this problem with @bbrockbernd , so perhaps he can do this.

Reply:
Let @bbrockbernd do this, sure